src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/PropertyPageDataProducer.cs

```diff
@@ -28,15 +28,27 @@ public static IEntityValue CreatePropertyPageValue(IEntityValue parent, IPropert
 
             var identity = new EntityIdentity(
                 ((IEntityWithId)parent).Id,
-                new KeyValuePair<string, string>[]
+                createKeys());
+
+            return CreatePropertyPageValue(parent.EntityRuntime, identity, cache, context, rule, debugChildRules, requestedProperties);
+
+            IEnumerable<KeyValuePair<string, string>> createKeys()
```

Comment:
>createKeys [](start = 54, length = 10)

What advantage does this have over the KeyValuePairs it's replacing? Is reading from the context expensive?

Reply:
No, the point here is that `context.ItemType` and `context.ItemName` may be null; if they are we don't want to add them to the key that identifies this entity in the Project Query API.